packages/postcss-minify-selectors/src/__tests__/index.js

```diff
@@ -505,6 +484,8 @@ test('should handle selectors from other plugins', () => {
   padding: 4px;
 }`;
   expect(
+    // eslint-disable-next-line no-warning-comments
+    // FIXME: not working as `toModules` plugin is not running properly
     postcss([toModules, plugin]).process(css, { from: undefined }).css
```

Comment:
@ai here, the `toModules` plugin is not working properly as its visitor keys are not being visited. Though the function is being called. Thats why the plugin is not transforming the selectors and the test case is failing

The definition for the plugin is [here](https://github.com/cssnano/cssnano/blob/9cfe8113ce30b4533bfc7be873e5c4055effb110/packages/postcss-minify-selectors/testPlugin/toModules.js)

Any suggestion why it is happening ?

Reply:
Everything looks good.

Can you debug this block of the code? https://github.com/postcss/postcss/blob/master/lib/lazy-result.js#L202-L213

Reply:
how ? the postcss dep is transpiled so cant really locate this one.

Reply:
What is in `postcss` dependency on this line? I think it should be `node_modules/postcss` in your project directory.

Open `node_modules/postcss/lib/lazy-result.js` in a text editor and put `console.log()` or use your favorite debugger. Try to look at what is in `this.listeners` (that PostCSS correctly cached `toModule` listener). Try to look at `this.plugins` to be sure that `toModule` was saved. `this.walkSync` calls will help you to understand what nodes were scanned.

Sorry, you need to help me. I am right now on other PostCSS 8 issue. 50M downloads lead to an insane number of edge cases.

Reply:
> What is in postcss dependency on this line? I think it should be node_modules/postcss in your project directory.

Yup I am checking this dir only, but the code is transpiled. 

![image](https://user-images.githubusercontent.com/26347874/94042022-a6267680-fde8-11ea-9b50-4af1fcd9d831.png)


is it postcss@7 ?

Reply:
It is PostCSS 7

Reply:
okay....looking into it.

Reply:
@ai  

here is the output

![image](https://user-images.githubusercontent.com/26347874/94043451-91e37900-fdea-11ea-9f15-4e0d5405842d.png)

Also, `toModules` has been registered. 

![image](https://user-images.githubusercontent.com/26347874/94043585-c2c3ae00-fdea-11ea-9b8a-bdd47b6e7e21.png)

Reply:
It means that an issue somewhere in that code https://github.com/postcss/postcss/blob/master/lib/lazy-result.js#L391-L422

Can you check how the algorithm is working there?

Reply:
yea, checking....

Reply:
Okay, I think some issue with normalizing the plugins. 

if the plugin type is function, should it be converted to object or should it be called before pushing to normalized plugins ?

cause [here in prepareVisitor](https://github.com/postcss/postcss/blob/master/lib/lazy-result.js#L392) my plugins are still functioning so the check would not be true and their wont be any listeners saved so the `this.hasListeners` check [here](https://github.com/postcss/postcss/blob/master/lib/lazy-result.js#L350) will fail and the plugin visitor keys wont be call.

I changed [this](https://github.com/postcss/postcss/blob/master/lib/processor.js#L50) to `normalized.push(i())` and now the visitor keys are working (not sure about any side effect for this).

Reply:
Seems like you forget to set `module.exports.postcss = true` and this why this line was not called https://github.com/postcss/postcss/blob/master/lib/processor.js#L40

Reply:
its there already, right ! .

[`plugin`](https://github.com/cssnano/cssnano/pull/954/files#diff-20446d189f703b4719024d8350448ef9R219)
[`toModules`](https://github.com/cssnano/cssnano/pull/954/files#diff-95973145503262cae31de6e4b30735acR26)

Reply:
Named export do not work like a property. You,  need to set `pluginCreator.postcss = true`, add a property to default exported function

Reply:
ohh, let me change this then